docs/csharp/programming-guide/unsafe-code-pointers/pointer-types.md

```diff
@@ -1,140 +1,96 @@
+- Any [enum](../../language-reference/keywords/enum.md) type.
+- Any pointer type.
+- Any user-defined struct type that contains fields of unmanaged types only.
+
+Pointer types do not inherit from [object](../../language-reference/keywords/object.md) and no conversions exist between pointer types and `object`. Also, boxing and unboxing do not support pointers. However, you can convert between different pointer types and between pointer types and integral types.
+
+When you declare multiple pointers in the same declaration, the asterisk (*) is written together with the underlying type only; it is not used as a prefix to each pointer name. For example:
+
+```csharp
+int* p1, p2, p3;   // Ok
+int *p1, *p2, *p3;   // Invalid in C#
+```
+
+A pointer cannot point to a reference or to a [struct](../../language-reference/keywords/struct.md) that contains references, because an object reference can be garbage collected even if a pointer is pointing to it. The garbage collector does not keep track of whether an object is being pointed to by any pointer types.
+
+The value of the pointer variable of type `myType*` is the address of a variable of type `myType`. The following are examples of pointer type declarations:
+
+|Example|Description|
+|-------------|-----------------|
+|`int* p`|`p` is a pointer to an integer.|
+|`int** p`|`p` is a pointer to a pointer to an integer.|
+|`int*[] p`|`p` is a single-dimensional array of pointers to integers.|
+|`char* p`|`p` is a pointer to a char.|
+|`void* p`|`p` is a pointer to an unknown type.|
+
+The pointer indirection operator * can be used to access the contents at the location pointed to by the pointer variable. For example, consider the following declaration:
+
+```csharp
+int* myVariable;
+```
+
+The expression `*myVariable` denotes the `int` variable found at the address contained in `myVariable`.
+
+There are several examples of pointers in the topics [fixed Statement](../../language-reference/keywords/fixed-statement.md) and [Pointer Conversions](../../programming-guide/unsafe-code-pointers/pointer-conversions.md). The following example shows the need for the `unsafe` keyword and the `fixed` statement, and how to increment an interior pointer.  You can paste this code into the Main function of a console application to run it. (Remember to enable unsafe code in the **Project Designer**; choose **Project**, **Properties** on the menu bar, and then select **Allow unsafe code** in the **Build** tab.)
```

Comment:
- shows the need for --> uses
- , and how to --> and shows how to
Outside of Visual Studio?

Reply:
One more place where the link to `-unsafe` page would be enough. Or include file that might be reused in all topics that show examples of `unsafe` code.

Reply:
I liked the link to -unsafe option.